### PR TITLE
URL rewrites are generated for categories with "Include in Menu=No" when creating a store view

### LINKED ISSUE #28358: URL rewrites are not generated for categories with "Include in Menu=No" when creating a store view
### Preconditions (*)

1. Magento 2.4-develop

### Steps to reproduce (*)

1. Navigate to Admin Panel > Catalog > Categories, select root category and add new subcategory:
```
Enable Category = Yes
Include in Menu = No
Category Name = Hidden Category
URL Key = hidden-category
```
2. Open created category on the frontend using direct URL /hidden-category.html
3. Navigate to Admin Panel > Stores > All Stores and create new store view:
```
Name: German
Code: german
Status: Enabled
```
4. Return to frontend, switch to German store view.
5. Open hidden category using direct URL /hidden-category.html

### Expected result (*)

1. Hidden category should be accessible on German store view.

### Actual result (*)

1. Hidden category is inaccessible on German store view, because URL rewrites are not generated for categories with "Include in Menu=No" when creating a store view.

---

- [ ] Severity: **S2**

### PR DESCRIPTION
**Pull request by engcom-Echo**
### Description (*)
Fix issue

### Fixed Issues (if relevant)

1. Fixes magento/magento2#28358

### Manual testing scenarios (*)
See magento/magento2#28358

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run all test

**Comment by engcom-Echo** [comment]
@magento run all test

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run all tests

File: app/code/Magento/Catalog/Model/Category.php
Comment (aleron75) [inline app/code/Magento/Catalog/Model/Category.php]: Hello, thank you for your contribution.

Reading the parameter name `$includeMenu` I was led to interpret the code as:
- `true` - select **only** categories that are **included** in the menu
- `false` - select **only** categories that are **not included** in the menu

Reading the code, instead, the real meaning of the two cases is:
- `true` - select **only** categories that are **included** in the menu (same as above)
- `false` - select **all categories**, regardless of the fact they are included in the menu or not

Doesn't `$onlyIncludeInMenu` sound more appropriate?

Cheers.

Comment (engcom-Echo) [reply app/code/Magento/Catalog/Model/Category.php]: Hi, @aleron75. Oh, why not - great. Thanks.

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by aleron75** [comment]
Hello @engcom-Echo,
there is a failing functional test that seems related to changes made by this PR, can you please take a look at it? I'll investigate the semantic checker issue instead.
Thank you!

**Comment by engcom-Echo** [comment]
Hi, @aleron75 failing functional test not related to PR, see first commit.
Thank you for your collaboration.

**Comment by aleron75** [comment]
I asked about the semantic checker, this PR adds new properties and method parameters that will require approval from Magento architects. I'll approve this PR but it doesn't depend only on me.
Thanks for your contribution so far!

**Review by aleron75 — DISMISSED** [review]
I am approving since failing functional test doesn't see related to this PR but since this PR changes some class contracts (see semantic checker complaints), Magento architects will have to make a final decision.

**Comment by magento-engcom-team** [comment]
Hi @aleron75, thank you for the review.
[ENGCOM-7726](https://jira.corp.magento.com/browse/ENGCOM-7726) has been created to process this Pull Request

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA passed
**Manual testing scenario:**

- Navigate to Admin Panel > Catalog > Categories, select root category and add new subcategory:
```
Enable Category = Yes
Include in Menu = No
Category Name = Hidden Category
URL Key = hidden-category
```
image
- Open created category on the frontend using direct URL `/hidden-category.html`
- Navigate to Admin Panel > Stores > All Stores and create new store view:
```
Name: German
Code: german
Status: Enabled
```
- Return to frontend, switch to German store view.
- Open hidden category using direct URL `/hidden-category.html`

:x: **Before:**
- Hidden category is inaccessible on German store view
image

:heavy_check_mark: **After:**

- Hidden category accessible on German store view.

image

**Comment by engcom-Charlie** [comment]
It needs to approve because there are minor failures in **Semantic Version Checker**.

File: app/code/Magento/Catalog/Model/Category.php
Comment (sidolov) [inline app/code/Magento/Catalog/Model/Category.php:984]: Unfortunately, we cannot change the signature of public methods in classes marked as @api according to our  [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.4/contributor-guide/backward-compatible-development/)

**Review by sidolov — CHANGES_REQUESTED** [review]

**Comment by engcom-Echo** [comment]
Unfortunately, I have to close this PR. Further development does not make sense.

**Pull request closed without merging**
